modules/capi-yaml-machine-template-azure.adoc

```diff
@@ -0,0 +1,46 @@
+// Module included in the following assemblies:
+//
+// * machine_management/cluster_api_machine_management/cluster_api_provider_configurations/cluster-api-config-options-azure.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="capi-yaml-machine-template-azure_{context}"]
+= Sample YAML for a Cluster API machine template resource on {azure-full}
+
+The machine template resource is provider-specific and defines the basic properties of the machines that a compute machine set creates.
+The compute machine set references this template when creating machines.
+
+[source,yaml]
+----
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureMachineTemplate # <1>
+metadata:
+  name: <template_name> # <2>
+  namespace: openshift-cluster-api
+spec:
+  template:
+    spec: # <3>
+      disableExtensionOperations: true
+      identity: UserAssigned
+      image:
+        id: /subscriptions/<subscription_id>/resourceGroups/<cluster_id>-rg/providers/Microsoft.Compute/galleries/gallery_<cluster_id>/images/<cluster_id>-gen2/versions/latest # <4> 
```

Comment:
Strictly speaking, this is not `gallery_<cluster_id>`. If `<cluster_id>` is `zhsun-fault-2m2np`, here is `gallery_zhsun_fault_2m2np`

Reply:
Good catch. There is a `_` instead of `-` on the gallery name.

Reply:
Change to <cluster_id> to <cluster_name> to not be different.

Reply:
Making sure I understand: if the cluster ID is `zhsun-fault-2m2np`, some transformation makes the value in the URL into `zhsun_fault_2m2np` instead?

Reply:
this is only true for the gallery_<value> part